syntax = "proto3";

import "google/protobuf/timestamp.proto";

message PBUserLogin {
  string user_name = 1;
}

message PBChatMessage {
  google.protobuf.Timestamp sent_at = 1;
  string from_user = 2;
  string text = 3;
}

message PBChatCommand {
  string command = 1;
  string parameter = 2;
}

message PBCommandResult {
  string command = 1;
  repeated string text = 2;
}

message PBMessage {
  oneof payload {
    PBUserLogin login = 1;
    PBChatMessage chat = 2;
    PBChatCommand command = 3;
    PBCommandResult result = 4;
  }
}
